frontend/src/Teacher/InbusImport.vue

```diff
@@ -0,0 +1,279 @@
+<script setup lang="ts">
+/**
+ * This component provides import of classes of students from Edison
+ * into Kelvin using the INBUS API.
+ *
+ * It is only available to teachers, and it is accessible from the main page
+ * next to class selection drop down menu.
+ */
+import { computed, onMounted, ref } from 'vue';
+
+import { csrfToken } from '../api.js';
+import { ConcreteActivity, InbusSubjectVersion } from './inbusdto';
+
+interface KelvinSubject {
+  name: string;
+  abbr: string;
+}
+
+interface Semester {
+  pk: number;
+  year: number;
+  winter: boolean;
+}
+
+interface ImportResult {
+  login: string;
+  firstname: string;
+  lastname: string;
+  created: boolean;
+}
+
+interface Result {
+  users: ImportResult[];
+  count: number;
+  error?: string;
+}
+
+let subject_inbus_selected = ref<InbusSubjectVersion | null>(null);
+let subject_kelvin_selected = ref(null);
+
+let semester = ref(null);
+let busy = ref<boolean>(false);
+
+let semesters = ref(null);
+
+let semesters_loading = ref<boolean>(true);
+let inbus_and_kelvin_subjects_loading = ref<boolean>(true);
+let inbus_schedule_loaded = ref<boolean>(false);
```

Comment:
You can just check if `subject_inbus_schedule` is `null` (and reset it to `null` at the beginning of `loadScheduleForSubjectVersionId`, no need to create another variable for it.

Reply:
Done.